### PR TITLE
Finish the second module of phoenix

### PR DESCRIPTION
**Pull request by samu975**

### CODE REVIEW

File: Phoenix/hello/lib/hello_web/controllers/hello_controller.ex [hunk 1]
```diff
@@ -0,0 +1,13 @@
+defmodule HelloWeb.HelloController do
+  use HelloWeb, :controller
+
+  def index(conn, _params) do
+    render(conn, "index.html")
+  end
```
Comment (juanhurtado10) [inline Phoenix/hello/lib/hello_web/controllers/hello_controller.ex:6]: @samu975 missing blank line

File: Phoenix/hello/lib/hello_web/controllers/page_controller.ex [hunk 2]
```diff
@@ -2,6 +2,10 @@ defmodule HelloWeb.PageController do
   use HelloWeb, :controller
 
   def index(conn, _params) do
-    render(conn, "index.html")
+    conn
```
Comment (juanhurtado10) [inline Phoenix/hello/lib/hello_web/controllers/page_controller.ex:5]: @samu975 what a `conn` is?

File: Phoenix/hello/lib/hello_web/endpoint.ex [hunk 3]
```diff
@@ -47,4 +47,17 @@ defmodule HelloWeb.Endpoint do
   plug Plug.Head
   plug Plug.Session, @session_options
   plug HelloWeb.Router
+
+  plug :introspect
```
Comment (juanhurtado10) [inline Phoenix/hello/lib/hello_web/endpoint.ex:51]: @samu975 Could you explain how a plug works?

File: Phoenix/hello/lib/hello_web/router.ex [hunk 4]
```diff
@@ -18,6 +19,15 @@ defmodule HelloWeb.Router do
     pipe_through :browser
 
     get "/", PageController, :index
+    get "/hello", HelloController, :index
+    get "/hello/:messenger", HelloController, :show
+    resources "/users", UserController
```
Comment (juanhurtado10) [inline Phoenix/hello/lib/hello_web/router.ex:24]: @samu975 What do resources mean?

File: Phoenix/hello/lib/hello_web/templates/hello/index.html.heex [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<section class="phx-hero">
+    <h2>Hello World, from Phoenix!</h2>
+</section>
```
Comment (juanhurtado10) [inline Phoenix/hello/lib/hello_web/templates/hello/index.html.heex:3]: @samu975 missing blank line at the end of the file. TIP: You can configure your text editor to handle this automatically for you!

File: Phoenix/hello/lib/hello_web/views/layout_view.ex [hunk 6]
```diff
@@ -3,5 +3,8 @@ defmodule HelloWeb.LayoutView do
```
Comment (juanhurtado10) [inline Phoenix/hello/lib/hello_web/views/layout_view.ex:3]: @samu975 what does a view do?

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]

File: Phoenix/hello/lib/hello_web/router.ex (hunk 4 above)
Comment (samu975) [reply Phoenix/hello/lib/hello_web/router.ex:24]: resouces is a macro. We use the resouces macro to add common routes and implement common actions like create, read, update and delete (crud)

File: Phoenix/hello/lib/hello_web/endpoint.ex (hunk 3 above)
Comment (samu975) [reply Phoenix/hello/lib/hello_web/endpoint.ex:51]: There exist two kinds of plugs, module plugs and function plugs.  A module plug is a module with two functions with some configurations. Both plugs work the same.  It receives a connections and options, transforms a connection and return the connection transformed.

File: Phoenix/hello/lib/hello_web/controllers/page_controller.ex (hunk 2 above)
Comment (samu975) [reply Phoenix/hello/lib/hello_web/controllers/page_controller.ex:5]: conn is a abbreviation for connection. Is an elixir struct filled with information regarding a web request like port, adapter, request_path, host, etc.

File: Phoenix/hello/lib/hello_web/views/layout_view.ex (hunk 6 above)
Comment (samu975) [reply Phoenix/hello/lib/hello_web/views/layout_view.ex:3]: Views are modules responsible for rendering. handles the structures data from the controller and converts it to a presentation to be shown to user

**Review by juanhurtado10 — APPROVED** [review]

**Pull request merged**